addons/pos_gift_card/models/gift_card.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class GiftCard(models.Model):
+    _inherit = "gift.card"
+
+    buy_pos_order_line_id = fields.Many2one(
+        "pos.order.line",
+        copy=False,
+        readonly=True,
+        help="Pos Order line where this gift card has been bought.",
+    )
+    redeem_pos_order_line_ids = fields.One2many(
+        "pos.order.line", "gift_card_id", string="Pos Redeems"
+    )
+
+    @api.model
+    def _generate_code(self):
```

Comment:
why overriding _generate_code? Why should we have different codes if we don't have POS or not.

Use the same code everywhere